cirq/testing/asynchronous.py

```diff
@@ -106,7 +105,11 @@ def assert_asyncio_will_raise(
             a matching exception.
     """
     try:
-        with pytest.raises(expected, match=match):
-            _run_loop_waiting_for(future, timeout)
+        _run_loop_waiting_for(future, timeout)
+    except expected as exc:
+        if match and not re.search(match, str(exc)):
+            assert False, "Pattern '{}' not found in '{}'".format(match, exc)
     except asyncio.TimeoutError:
         assert False, "Not done: {!r}".format(future)
+    else:
```

Comment:
This works?!

Reply:
Yup :-) Can also have `else:` clause on for-loops, which will execute if the loop completes without `break`ing (that one is a bit obscure, but occasionally very useful).